src/app/CommandHandlerInterfaceRegistry.cpp

```diff
@@ -0,0 +1,139 @@
+/**
+ *    Copyright (c) 2024 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include <app/CommandHandlerInterfaceRegistry.h>
+
+using namespace chip::app;
+
+namespace {
+
+CommandHandlerInterface * gCommandHandlerList = nullptr;
+
+}
+
+namespace chip {
+namespace app {
+namespace CommandHandlerInterfaceRegistry {
```

Comment:
So this makes it impossible to have separate registries, if we ever get to the point where we can run multiple stacks in the same process.

That's probably fine for now, but just something to watch out for.  Might have been better if there were an actual object, not a namespace, so if we ever get to a point where we have a better way to get this object than just calling a global function there would be fewer updates to consumers.

Reply:
My objective is to decouple the CommandHandlerInterfaceRegistry/list from the InteractionModelEngine and place it to be an implementation detail of the data model handling.

After that, the IME can have its own data-model and we can run things in parallel. Agree that this is a step backwards right now ... unfortunate sideffect to prepare to have full decoupling for later. Right now we are not able to parallel run properly anyway because AttributeAccessInterface is singleton-based.

Reply:
> Agree that this is a step backwards right now 

My point is that this is an entirely avoidable step backwards, no?

Reply:
Discussed offline: this should be symmetric with AAI, but AAI choice is already bad just because legacy. We will create objects with singletons as a followup on both.